python/vuln_details/blog_get_vuln_details.py

```diff
@@ -0,0 +1,337 @@
+        print_error("The file's format is probably not JSONL, but XML or CSV")
+        sys.exit(1)
+    
+    return vuln
+
+if __name__ == "__main__":
+    logging_file_name = "get_vuln_details.log"
+    logging_format = "%(asctime)s %(levelname)s %(message)s"
+    logging.basicConfig(filename=logging_file_name, level=logging.INFO, format=logging_format)
+    print_info("Vulnerabiity Exports with Details")
+
+    # Process command line arguments.
+    export_id = 0
+    try:
+        if len(sys.argv) == 2:
+            if sys.argv[1] == "-h":
+                print_help()
+            else:
+                export_id = int(sys.argv[1])
+    except ValueError:
+        print("Bad string to integer conversion for search ID")
+        print_help()
+    
+    # Obtain the Kenna Security API key.
+    api_key = os.getenv('KENNA_API_KEY')
+    if api_key is None:
+        print_error("Environment variable KENNA_API_KEY is non-existent")
+        sys.exit(1)
+    
+    # HTTP headers for Kenna.
+    headers = {'Accept': 'application/json',
+               'Content-Type': 'application/json; charset=utf-8',
+               'X-Risk-Token': api_key,
+               'User-Agent': 'vuln_exports/1.0.0 (Kenna Security)'}
+
+    # You might have to change this depending on your deployment.
+    base_url = "https://api.kennasecurity.com"
+    
+    # Seleted fields to be returned.
+    selected_fields = [
+        "cve_description", 
+        "cve_id",
+        "description",
+        "details", 
+        "id",
+        "scanner_score",
+        "solution",
+        "risk_meter_score", 
+    ]
+
+    # If an export ID is not defined then do an vuln export.
+    if export_id == 0:
+        export_id = request_vuln_exports(base_url, headers, selected_fields)
+    else:
+        print_info(f"Using search ID: {export_id}")
+    check_export_status(base_url, headers, export_id)
+    
+    # Create the file name based on the search ID.
+    vuln_file_name = f"vulns_{export_id}"
+    jsonl_vuln_file_name = retrieve_vuln_data(base_url, headers, export_id, vuln_file_name)
+    
+    # Count and report.
+    num_vulns = count_lines(jsonl_vuln_file_name) 
+    if num_vulns == 1:
+        print_error(f"The format of file {jsonl_vuln_file_name} is probably JSON, not JSONL")
+        sys.exit(1)
+
+    print_info(f"File: {jsonl_vuln_file_name} with {num_vulns} vulns.")
+
+    file_name_without_ext = os.path.splitext(jsonl_vuln_file_name)[0]
+    json_vuln_file_name = f"{file_name_without_ext}.json"
+    html_vuln_file_name = f"{file_name_without_ext}.html"
+
+    # Initialize the HTML Sanitizer and open the output files.
+    sanitizer = Sanitizer()  # default configuration
+    json_f = open(json_vuln_file_name, 'w')
+    html_f = open(html_vuln_file_name, 'w')
+
+    # Read the JSONL input file and write out pretty JSON and HTML files.
+    num_lines = 0
+    with open(jsonl_vuln_file_name, 'r') as jsonl_f:
+        for line_num, vuln_line in enumerate(jsonl_f):
```

Comment:
Receiving an error here. Likely due to crappy data in the details field. Additional sanitization may be needed.

`(pandas_env) PS C:\Users\jaredk\OneDrive - Cisco\Documents\Scripts\vuln_details_export> python blog_get_vuln_details.py Vulnerabiity Exports with Details
New export search ID: 2909784 with 4607 vulnerabilities.
Sleeping for 5 seconds.  (16 out of 240)
Gunzipping file vulns_2909784.gz to vulns_2909784.jsonl
File vulns_2909784.gz gunzipped to vulns_2909784.jsonl
Counting lines in vulns_2909784.jsonl
File: vulns_2909784.jsonl with 4607 vulns.
Traceback (most recent call last):
  File "C:\Users\jaredk\OneDrive - Cisco\Documents\Scripts\vuln_details_export\blog_get_vuln_details.py", line 318, in <module>
    for line_num, vuln_line in enumerate(jsonl_f):
  File "C:\Users\jaredk\Miniconda3\envs\pandas_env\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 7268: character maps to <undefined> `

Reply:
this is going to be a problem with details from json export and it still exists in Scanner Vulnerability Details API endpoint as well.